scripts/generate_function_json.py

```diff
@@ -0,0 +1,66 @@
+import json
+from pathlib import Path
+from argparse import ArgumentParser
+from subprocess import check_output, CalledProcessError, PIPE
+
+parser = ArgumentParser()
+parser.add_argument('--source', required=True)
+parser.add_argument('--binary', required=True)
+args = parser.parse_args()
+
+
+def get_result(example: str) -> str:
+    try:
+        out = check_output(
+            [
+                args.binary,
+                '-json',
+                '-c',
+                'SELECT setseed(0.42); ',
+                '-c',
+                'LOAD icu;',
+                '-c',
+                f'SELECT {example.strip(";")} AS result'
+            ],
+            stderr=PIPE
+        )
+        rows = json.loads(out.splitlines()[-1])
+        return rows[0]['result']
+    except CalledProcessError as e:
+        print(example.strip(), e.stderr.decode())
+        return None
+
+
+def main():
+    functions = []
+    source = Path(args.source)
+    assert source.exists()
+    for file in source.glob('src/core_functions/**/*.json'):
+        category = file.parent.stem
+        with file.open() as fh:
+            functions += [
+                {
+                    **function,
+                    'parameters': (
+                        function['parameters'].split(',')
+                        if function.get('parameters')
+                        else
+                        []
+                    ),
+                    'category': category,
+                    'result': get_result(function['example']) if function.get('example') else None
+                }
+                for function in json.load(fh)
+            ]
+
+    functions = sorted(functions, key=lambda fn: fn['name'])
+
+    with open('docs/functions.json', 'w') as fh:
```

Comment:
One small addition I would consider (even later) is given that we need a binary, we can add a field (either to `docs/functions.json` or some ancillary file) like: "duckdb_version": "some_git_hash". I don't see a strong use right now but for avoiding re-generating if fields is the same or tracking down weird behavioiur, but I guess the extra cost is low.

Reply:
Given the way this stuff works is that we don't commit the new file if it hasn't changed, maybe.

But yeah, the fact that the times in this file change on each run is annoying